GVFS/GVFS.UnitTests/Common/SHA1UtilTests.cs

```diff
@@ -74,5 +74,16 @@ public void IsValidFullSHATrueForLength40HexStrings()
             SHA1Util.IsValidShaFormat("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa").ShouldEqual(true);
             SHA1Util.IsValidShaFormat("ffffffffffffffffffffffffffffffffffffffff").ShouldEqual(true);
         }
+
+        [TestCase]
+        public void IsValidHex()
+        {
+            SHA1Util.IsValidHex("1234567890123456789012345678901234567890", 40).ShouldEqual(true);
+            SHA1Util.IsValidHex("abcdef7890123456789012345678901234567890", 20).ShouldEqual(false);
+            SHA1Util.IsValidHex("ab", 2).ShouldEqual(true);

```

Comment:
What happens if our string has a different length than the second parameter? This currently doesn't make it clear what should happen.